### DISCUSSION TITLE
Where can I read about colorScheme?

### DISCUSSION THREAD

**Discussion by ghost** [chakra-ui/chakra-ui#2846] [Q&A] [answered]
Several of the components support the colorScheme property but I can't find any references to colorScheme in the documentation.

No list of colors or how to use the property.

Can I get some guidance?

**Comment by chr-ge** [comment, marked as answer]
> The `colorScheme` prop represents a visual color scheme, not a css color attribute. 

You pass it a color name such as "blue", "green", "red", etc. You can pass it any of the default theme colors except "black" and "white"
If you give a `Button` component, for example, a `colorScheme` of "red", it uses the different palette ranges for its color and `hover`, `disabled`, `active` pseudo-classes.

**Reply by VikaramjitSinghGorraya** [reply]
what is the difference between using colorScheme and color?

**Reply by chr-ge** [reply]
Answerd here: https://github.com/chakra-ui/chakra-ui/discussions/2961

**Comment by davidparys** [comment]
So an additional question would be is there a way to add a custom color scheme, I'd like to design let's say a button with all the background, borders & font colors on active, hover etc. I haven't seen any documentation about that

**Reply by chr-ge** [reply]
To create a custom `colorScheme` just add a color with a range of palettes to your theme. 
You can use websites like [Smart Swatch](https://smart-swatch.netlify.app/), [Coolors](https://coolors.co/app) or [Palx](https://palx.jxnblk.com/) to generate a palette.
```js
import {  extendTheme} from "@chakra-ui/react";

const theme = extendTheme({
  colors: {
    lime: {
      50: '#f2ffde',
     100: '#defcb2',
     200: '#caf884',
     300: '#b5f554',
     400: '#a1f226',
     500: '#88d90d',
     600: '#69a905',
     700: '#4a7801',
     800: '#2b4800',
     900: '#0b1900',
    }
  }
})
```

**Reply by x-EricH-x** [reply]
Hi, following the question, is there a way to customize (/ which source file I should look into) if I want to change/know which color is used in which part of the components (under a color scheme)?
I am trying to customize the `Tag` and `Button` component. The source file of `Tag` component does not reference any `x00` color code (e.g. `100`, `200`, etc), it only references the `colorScheme`, so the exact relationship is defined elsewhere.
https://github.com/chakra-ui/chakra-ui/blob/724d3e15100324be88d74ecf664356ed9f8205c6/packages/theme/src/components/tag.ts#L94

Edit:
I noticed this question is answered (so better remain closed), so I opened a new question instead:
https://github.com/chakra-ui/chakra-ui/discussions/3737

**Reply by henriqgoncalvs** [reply]
> To create a custom `colorScheme` just add a color with a range of palettes to your theme.
> You can use websites like [Smart Swatch](https://smart-swatch.netlify.app/), [Coolors](https://coolors.co/app) or [Palx](https://palx.jxnblk.com/) to generate a palette.
> 
> ```js
> import {  extendTheme} from "@chakra-ui/react";
> 
> const theme = extendTheme({
>   colors: {
>     lime: {
>       50: '#f2ffde',
>      100: '#defcb2',
>      200: '#caf884',
>      300: '#b5f554',
>      400: '#a1f226',
>      500: '#88d90d',
>      600: '#69a905',
>      700: '#4a7801',
>      800: '#2b4800',
>      900: '#0b1900',
>     }
>   }
> })
> ```

Thanks man!!! That helped so so much. Didn't know Palx and it fited like a glove for my problem

**Reply by estebandinamarca** [reply]
Thanks, but how can I customize the behavior of color states, for example, to use `lime.900` instead of the default color for "hover"?

**Reply by blazyy** [reply]
How would we go about specifiying color schemes for dark mode? They seem to be lost when shifting to dark mode

**Comment by remarsh86** [comment]
I had the same issue and found a detailed description of colorScheme here: [https://chakra-ui.com/docs/form/button#custom-button](url)

List of colors:
colorScheme
Type
"whiteAlpha" | "blackAlpha" | "gray" | "red" | "orange" | "yellow" | "green" | "teal" | "blue" | "cyan" | "purple" | "pink" | "linkedin" | "facebook" | "messenger" | "whatsapp" | "twitter" | "telegram"
Default
"gray"

**Reply by airandfingers** [reply]
Getting a 404 for that URL. New one: https://chakra-ui.com/docs/form/button#button-props

**Comment by Krishna698-web** [comment]
Hello there, I'm getting the following error on my application, can anyone please help me out with this??
Cannot read properties of undefined (reading 'colorScheme')
TypeError: Cannot read properties of undefined (reading 'colorScheme')

**Comment by sooohka** [comment]
Where can i read information about generating type information about custom colorScheme?

**Reply by junkieshuffle** [reply]
https://v2.chakra-ui.com/docs/styled-system/cli
